AK/IPv4Address.h

```diff
@@ -76,12 +76,34 @@ class [[gnu::packed]] IPv4Address
         if (string.is_null())
             return {};
         auto parts = string.split_view('.');
-        if (parts.size() != 4)
-            return {};
-        auto a = parts[0].to_uint().value_or(256);
-        auto b = parts[1].to_uint().value_or(256);
-        auto c = parts[2].to_uint().value_or(256);
-        auto d = parts[3].to_uint().value_or(256);
+
+        u32 a;
+        u32 b;
+        u32 c;
+        u32 d;
+
+        if (parts.size() == 1) {
+            a = 0;
+            b = 0;
+            c = 0;
+            d = parts[1].to_uint().value_or(256);
+        } else if (parts.size() == 2) {
+            a = parts[1].to_uint().value_or(256);
+            b = 0;
+            c = 0;
+            d = parts[2].to_uint().value_or(256);
+        } else if (parts.size() == 3) {
+            a = parts[0].to_uint().value_or(256);
+            b = parts[1].to_uint().value_or(256);
+            c = 0;
+            d = parts[2].to_uint().value_or(256);
+        } else {
+            a = parts[0].to_uint().value_or(256);
+            b = parts[1].to_uint().value_or(256);
+            c = parts[2].to_uint().value_or(256);
+            d = parts[3].to_uint().value_or(256);
+        }
```

Comment:
I think this should probably be another `else if (parts.size() == 4)` and then a final `else` returning `{}`? Otherwise this would silently accept and ignore more than 4 parts, no? (1.2.3.4.5.6)

Reply:
True, will correct.

Reply:
Corrected in 70b7687.